tool/cmd/main.go

```diff
@@ -4,58 +4,58 @@
 package main
 
 import (
-	"fmt"
-	"io"
 	"log/slog"
 	"os"
+	"path/filepath"
 	"time"
 
 	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/tool/ex"
-	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/tool/internal/instrument"
-	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/tool/internal/setup"
 	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/tool/util"
+	"github.com/urfave/cli/v2"
 )
 
 const (
-	ActionSetup      = "setup"
-	ActionGo         = "go"
-	ActionIntoolexec = "toolexec"
-	ActionVersion    = "version"
-	DebugLog         = "debug.log"
+	exitCodeFailure    = -1
+	exitCodeUsageError = 2
 )
 
-func cleanBuildTemp() {
-	_ = os.RemoveAll(setup.OtelRuntimeFile)
+func main() {
+	app := cli.App{
+		Name:        "otel",
+		Usage:       "OpenTelemetry Go Compile-Time instrumentation tool",
+		HideVersion: true,
+		Flags: []cli.Flag{
+			&cli.PathFlag{
+				Name:      "work-dir",
+				Aliases:   []string{"w"},
+				Usage:     "The path to a directory where working files will be written",
+				TakesFile: false,
+				Value:     filepath.Join(".", util.BuildTempDir),
+			},
+		},
+		Commands: []*cli.Command{
+			commandSetup,
+			commandGo,
+			commandToolexec,
+			commandVersion,
+		},
+		Before: initLogger,
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		panic(err)
+	}
 }
 
-func initLogger(phase string) (*slog.Logger, error) {
-	var writer io.Writer
-	switch phase {
-	case ActionSetup, ActionGo:
-		// Create .otel-build dir
-		_, err := os.Stat(util.BuildTempDir)
-		if os.IsNotExist(err) {
-			err = os.MkdirAll(util.BuildTempDir, 0o755)
-			if err != nil {
-				return nil, ex.Errorf(err, "failed to create .otel-build dir")
-			}
-		}
-		// Configure slog to write to the debug.log file
-		path := util.GetBuildTemp(DebugLog)
-		logFile, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o777)
-		if err != nil {
-			return nil, ex.Errorf(err, "failed to open log file")
-		}
-		writer = logFile
-	case ActionIntoolexec:
-		path := util.GetBuildTemp(DebugLog)
-		logFile, err := os.OpenFile(path, os.O_WRONLY|os.O_APPEND, 0o777)
-		if err != nil {
-			return nil, ex.Errorf(err, "failed to open log file")
-		}
-		writer = logFile
-	default:
-		return nil, ex.Errorf(nil, "invalid action: %s", phase)
+func initLogger(cCtx *cli.Context) error {
+	buildTempDir := cCtx.Path("work-dir")
+	if err := os.MkdirAll(buildTempDir, 0o755); err != nil {
```

Comment:
For consistency with the rest of the codebase, could you please avoid using the if short statement? The established pattern here is to declare the variable on a separate line.